test/connection_string/test/valid-options.json

```diff
@@ -37,6 +37,25 @@
       "options": {
         "tls": true
       }
+    },
+    {
+      "description": "Colon in a key value pair",
+      "uri": "mongodb://example.com?authMechanismProperties=TOKEN_RESOURCE:mongodb://test-cluster",
+      "valid": true,
+      "warning": false,
+      "hosts": [
+        {
+          "type": "hostname",
+          "host": "example.com",
+          "port": null
+        }
+      ],
+      "auth": null,
+      "options": {
+        "authmechanismProperties": {
+          "TOKEN_RESOURCE": "mongodb://test-cluster"
+        }
+      }
```

Comment:
Why did a new a colon test get added?

Reply:
do we now need an XFAIL test with a comma in it?

Reply:
This came up because `TOKEN_RESOURCE` is the first auth mech property that will commonly have a colon in it, and that part of the spec was ambiguous.   I added a test for the case with a comma.